authenticator_sdk/src/main/kotlin/com/saltedge/authenticator/sdk/tools/keystore/KeyStoreManager.kt

```diff
@@ -225,6 +210,29 @@ object KeyStoreManager : KeyStoreManagerAbs {
         }
     }
 
+    /**
+     * Get RSA key pair by the given alias
+     *
+     * @param alias - the alias name
+     * @return KeyPair object
+     */
+    override fun getKeyPair(alias: String?): KeyPair? {
+        return try {
+            val keyAlias = alias ?: return null
+            val store = androidKeyStore ?: return null
+            (store.getKey(keyAlias, null) as? PrivateKey)?.let { privateKey ->
+                val publicKey: PublicKey? = store.getCertificate(keyAlias).publicKey
+                KeyPair(publicKey, privateKey)
+            }
+        } catch (e: UnrecoverableKeyException) {
+            androidKeyStore?.deleteEntry(alias)
```

Comment:
You should not delete entry if you can't read it.

Reply:
In our case, if we get into this branch(`UnrecoverableKeyException`) , then the only thing that should be done is to delete it.
Why do I need to store an entry if I can't read it?